src/Raven.Server/Commercial/LicenseHandler.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Imports.Newtonsoft.Json;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.BackgroundTasks;
+using Sparrow.Logging;
+
+namespace Raven.Server.Commercial
+{
+    public class LicenseHandler : IDisposable
```

Comment:
Also, we probably want a better name for it, `LicenseHandler` is already the name of the request handler.